wildlifelicensing/apps/returns/tests/test_returns.py

```diff
@@ -121,3 +112,144 @@ def test_lodge_return(self):
+            self.assertEqual(403, response.status_code)
+            self.client.logout()
+
+    def test_curate_view(self):
+        """
+        Only officer can curate returns
+        """
+        url = reverse('wl_returns:curate_return', args=(self.ret.pk,))
+        allowed = [self.officer]
+        forbidden = [self.not_allowed_customer, self.assessor, self.customer]
+
+        for user in allowed:
+            self.client.login(user.email)
+            response = self.client.get(url)
+            self.assertEqual(200, response.status_code)
+            self.client.logout()
+
+        for user in forbidden:
+            self.client.login(user.email)
+            response = self.client.get(url)
+            self.assertEqual(403, response.status_code)
+            self.client.logout()
+
+    def test_view_log(self):
+        """
+        Only officer can view log
+        """
+        url = reverse('wl_returns:log_list', args=(self.ret.pk,))
+        allowed = [self.officer]
+        forbidden = [self.not_allowed_customer, self.assessor, self.customer]
+
+        for user in allowed:
+            self.client.login(user.email)
+            response = self.client.get(url)
+            self.assertEqual(200, response.status_code)
+            self.client.logout()
+
+        for user in forbidden:
+            self.client.login(user.email)
+            response = self.client.get(url)
+            self.assertEqual(403, response.status_code)
+            self.client.logout()
+
+    def test_add_log(self):
+        """
+        Only officer can view log
+        """
+        url = reverse('wl_returns:add_log_entry', args=(self.ret.pk,))
+        allowed = [self.officer]
+        forbidden = [self.not_allowed_customer, self.assessor, self.customer]
+
+        payload = {
+            'to': 'user',
+            'from': 'test',
+            'type': 'email'
+        }
+        for user in allowed:
+            self.client.login(user.email)
+            response = self.client.post(url, data=payload)
+            self.assertEqual(200, response.status_code)
+            self.client.logout()
+
+        for user in forbidden:
+            self.client.login(user.email)
+            response = self.client.post(url, data=payload)
+            self.assertEqual(403, response.status_code)
+            self.client.logout()
+
+    def test_download_template(self):
+        """
+        Every authenticated user should be able to download a return template
+        """
+        return_type = self.ret.return_type
+        url = reverse('wl_returns:download_return_template', args=(return_type.pk,))
+        allowed = [self.officer,self.not_allowed_customer, self.assessor, self.customer]
+        forbidden = []
+
+        for user in allowed:
+            self.client.login(user.email)
+            response = self.client.get(url)
+            self.assertEqual(200, response.status_code)
+            self.client.logout()
+
+        for user in forbidden:
```

Comment:
because if the spec change and we want to forbid someone one day we just add it to the list, it is also a statement that says everyone is allowed.